sdks/python/apache_beam/runners/worker/sdk_worker.py

```diff
@@ -628,8 +632,14 @@ def _blocking_request(self, request):
       return response
 
   def _next_id(self):
-    self._last_id += 1
-    return str(self._last_id)
+    with self._lock:
```

Comment:
The lock is required since we are using simple integer: 

https://stackoverflow.com/questions/23547604/python-counter-atomic-increment
https://www.reddit.com/r/Python/comments/52mlee/a_threadsafe_incrementing_counter_very_simple_but/d7m6mcn/

This could be an alternative: https://pypi.org/project/atomiclong/

Reply:
Yes, the lock is required. In Java it would also be required for a regular integer. I'm not convinced an atomic long would make a difference here. The lock basically comes at no cost here. Like I pointed out in the upstream PR, there are major other factors which impact performance related to the parallelism model of the SDK Harness, i.e. the use of Python threads which, due to the GIL, results in single-threaded performance.